app/bundles/LeadBundle/Form/Type/CompanyMergeType.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * @copyright   2015 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Form\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/**
+ * Class CompanyMergeType.
+ */
+class CompanyMergeType extends AbstractType
+{
+    /**
+     * @param FormBuilderInterface $builder
+     * @param array                $options
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder->add(
+            'company_to_merge',
+            'company_list',
+            [
+                'multiple'    => false,
+                'required'    => true,
+                'modal_route' => false,
+                'main_entity' => $options['main_entity'],
+                'constraints' => [
+                    new NotBlank(
+                        ['message' => 'mautic.company.choosecompany.notblank']
+                    ),
+                ],
+            ]
+        );
+
+        $builder->add(
+            'buttons',
+            'form_buttons',
+            [
+                'apply_text' => false,
+                'save_text'  => 'mautic.company.merge',
```

Comment:
This is kind of awkward for the button (`Company merge`). Do we really need to specify `Company` or can we use the existing `mautic.lead.merge` string?